Tasks/AzurePowerShellV5/RemoveAzContext.ps1

```diff
@@ -0,0 +1,91 @@
+. "$PSScriptRoot/Utility.ps1"
+Update-PSModulePathForHostedAgentLinux 
+
+function CmdletHasMember {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$cmdlet,
+        [Parameter(Mandatory=$true)]
+        [string]$memberName)
+    try{
+        $hasMember = (gcm $cmdlet).Parameters.Keys.Contains($memberName)
+        return $hasMember
+    }
+    catch
+    {
+        return $false;
+    }
+}
+
+function Disconnect-UsingAzModule {
+    [CmdletBinding()]
+    param(
+        [string]$restrictContext = 'False'
+    )
+
+    if ((Get-Command -Name "Disconnect-AzAccount" -ErrorAction "SilentlyContinue") -and (CmdletHasMember -cmdlet Disconnect-AzAccount -memberName Scope)) {
+        if ($restrictContext -eq 'True') {
+            Write-Host "##[command]Disconnect-AzAccount -Scope CurrentUser -ErrorAction Stop"
+            $null = Disconnect-AzAccount -Scope CurrentUser -ErrorAction Stop
+        }
+        Write-Host "##[command]Disconnect-AzAccount -Scope Process -ErrorAction Stop"
+        $null = Disconnect-AzAccount -Scope Process -ErrorAction Stop
+    } 
+    if (Get-Command -Name "Clear-AzContext" -ErrorAction "SilentlyContinue") {
+        Write-Host "##[command]Clear-AzContext -Scope Process -ErrorAction Stop"
+        $null = Clear-AzContext -Scope Process -ErrorAction Stop
+    }
+}
+
+function Disconnect-UsingARMModule {
+    [CmdletBinding()]
+    param()
+
+    if ((Get-Command -Name "Disconnect-AzureRmAccount" -ErrorAction "SilentlyContinue") -and (CmdletHasMember -cmdlet Disconnect-AzureRmAccount -memberName Scope)) {
+        Write-Host "##[command]Disconnect-AzureRmAccount -Scope Process -ErrorAction Stop"
+        $null = Disconnect-AzureRmAccount -Scope Process -ErrorAction Stop
+    }
+    elseif ((Get-Command -Name "Remove-AzureRmAccount" -ErrorAction "SilentlyContinue") -and (CmdletHasMember -cmdlet Remove-AzureRmAccount -memberName Scope)) {
+        Write-Host "##[command]Remove-AzureRmAccount -Scope Process -ErrorAction Stop"
+        $null = Remove-AzureRmAccount -Scope Process -ErrorAction Stop
+    }
+    elseif ((Get-Command -Name "Logout-AzureRmAccount" -ErrorAction "SilentlyContinue") -and (CmdletHasMember -cmdlet Logout-AzureRmAccount -memberName Scope)) {
+        Write-Host "##[command]Logout-AzureRmAccount -Scope Process -ErrorAction Stop"
+        $null = Logout-AzureRmAccount -Scope Process -ErrorAction Stop
+    }
+
+    if (Get-Command -Name "Clear-AzureRmContext" -ErrorAction "SilentlyContinue") {
+        Write-Host "##[command]Clear-AzureRmContext -Scope Process -ErrorAction Stop"
+        $null = Clear-AzureRmContext -Scope Process -ErrorAction Stop
+    }
+}
+
+
+function Disconnect-AzureAndClearContext {
```

Comment:
Why don't you call the existing `Disconnect-AzureAndClearContext` from [utility.ps1](https://github.com/microsoft/azure-pipelines-tasks/blob/6e43ceb8bc63a92ceaa4b7c5b0013a329b1e91fa/Tasks/Common/VstsAzureHelpers_/Utility.ps1#L618)?

Reply:
We can't use that utility as it was written specific to windows and present on common which is not supportable for linux hence we created a new file.

Reply:
But the code you copied out from there is exactly the same. It would be worth to extract it to separate shared ps1 file and use it on both places.